src/overlays/actors/ovl_En_Dns/z_en_dns.c

```diff
@@ -27,74 +52,603 @@ const ActorInit En_Dns_InitVars = {
+                this->unk_2E4 -= -3.0769231f;
+            }
+        } else {
+            if (func_801378B8(&this->skelAnime, 0.0f) || func_801378B8(&this->skelAnime, 6.0f) ||
+                func_801378B8(&this->skelAnime, 13.0f)) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_WALK);
+            }
+
+            if (this->skelAnime.animCurrentFrame > 7.0f) {
+                this->unk_2E4 += -1.5384616f;
+            }
+        }
+        sp2C.x = this->unk_2E4;
+        sp2C.y = 0.0f;
+        sp2C.z = 0.0f;
+        Lib_Vec3f_TranslateAndRotateY(&this->actor.home.pos, this->actor.world.rot.y, &sp2C, &this->actor.world.pos);
+    }
+
+    return ret;
+}
+
+s32 func_8092D068(EnDns* this) {
+    s32 ret = false;
+
+    if (ENDNS_PARAMS_8000(&this->actor)) {
+        if (gSaveContext.weekEventReg[23] & 0x20) {
+            ret = true;
+        }
+    } else if (ENDNS_PARAMS_4000(&this->actor)) {
+        if ((gSaveContext.weekEventReg[9] & 0x80) && !(gSaveContext.weekEventReg[23] & 0x20)) {
+            ret = true;
+        }
+    } else if (!(gSaveContext.weekEventReg[9] & 0x80) && !(gSaveContext.weekEventReg[23] & 0x20)) {
+        ret = true;
+    }
+
+    return ret;
+}
+
+void func_8092D108(EnDns* this, GlobalContext* globalCtx) {
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_8012C28C(globalCtx->state.gfxCtx);
+
+    SysMatrix_SetStateRotationAndTranslation(this->actor.home.pos.x, this->actor.home.pos.y, this->actor.home.pos.z,
+                                             &this->actor.home.rot);
+    Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
+
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, D_06002C48);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_8092D1B8(EnDns* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s16 sp22 = this->actor.world.rot.y;
+
+    if (ENDNS_PARAMS_4000(&this->actor)) {
+        func_8092CE38(this);
+    }
+
+    if (!ENDNS_PARAMS_4000(&this->actor) || (this->unk_2D2 != 0)) {
+        if (!(gSaveContext.weekEventReg[23] & 0x20) && !(gSaveContext.eventInf[1] & 0x20) && func_8092CC68(globalCtx)) {
+            player->stateFlags1 |= 0x20;
+            this->unk_2C6 |= 0x100;
+            func_8013AED4(&this->unk_2C6, 4, 7);
+            play_sound(NA_SE_SY_FOUND);
+            gSaveContext.eventInf[1] |= 0x20;
+            this->unk_2F4 = func_8092CCEC;
+            func_8092C63C(this, 2);
+            this->actionFunc = func_8092D320;
+        } else if (gSaveContext.eventInf[1] & 0x40) {
+            func_8092CCEC(this, globalCtx);
+            func_8092C63C(this, 2);
+            this->actionFunc = func_8092D330;
+        }
+        Math_ApproachS(&this->actor.shape.rot.y, sp22, 3, 0x2AA8);
+    }
+}
+
+void func_8092D320(EnDns* this, GlobalContext* globalCtx) {
+}
```

Comment:
```suggestion
void EnDns_Noop(EnDns* this, GlobalContext* globalCtx) {
}
```
Might as well get this one out of the way.